src/read.c

```diff
@@ -1248,15 +1267,16 @@ void ReadFuncExpr (
         Match( S_SEMICOLON, ";", STATBEGIN|S_END|follow );
     }
 
-    /* 'function( ... arg )' takes a variable number of arguments              */
-    if (narg >= 1 && ! strcmp( "arg", CSTR_STRING( ELM_LIST(nams, narg) ) ) )
+    /* 'function( a,b, ...p )' takes a variable number of arguments           */
+    /* Also, we special case function(arg)                                   */
+    if (isvarg || ( narg >= 1 && ! strcmp( "arg", CSTR_STRING( ELM_LIST(nams, narg) ) )) )
       {
-	/* TODO: remove this before the next non-beta release */
-	if (narg > 1)
-	  SyntaxWarning("New syntax used -- intentional?");
 	narg = -narg;
       }
-	
+      
+    if ( narg > 1 && ! strcmp( "arg", CSTR_STRING( ELM_LIST(nams, narg) ) ) ) {
+      		SyntaxWarning("arg more arguments");
+      }
     /*     if ( narg == 1 && ! strcmp( "arg", CSTR_STRING( ELM_LIST(nams,1) ) ) )
```

Comment:
I guess this is just in here to flag places that already switched to the new-but-now-hopefully-soon-deprecated variadic `function(a,b,c,arg)`? If so, perhaps the message should be adjusted? Or perhaps we can remove it?
In particular, do we know of any users of this feature outside of the library (i.e. any packages)?